src/AbstractBlocklockReceiver.sol

```diff
@@ -1,20 +1,196 @@
+    /// @dev funding and consumer contract address registration.
+    uint256 public subscriptionId;
+
+    /// @notice Ensures that the caller is the designated Blocklock contract.
+    /// @dev This modifier restricts access to the function it modifies to only the Blocklock contract.
+    ///      If the caller is not the Blocklock contract, the transaction will revert with an error message.
+    /// @notice Reverts with the error message "Only blocklock contract can call" if the caller is not the Blocklock contract.
     modifier onlyBlocklockContract() {
-        require(msg.sender == address(blocklock), "Only timelock contract can call this.");
+        require(msg.sender == address(blocklock), "Only blocklock contract can call");
         _;
     }
 
-    constructor(address blocklockSender) {
+    constructor(address blocklockSender) ConfirmedOwner(msg.sender) {
         blocklock = IBlocklockSender(blocklockSender);
     }
 
-    function receiveBlocklock(uint256 requestID, bytes calldata decryptionKey) external virtual onlyBlocklockContract {}
+    /// @notice Receives a blocklock request and the associated decryption key.
+    /// @dev This function is only callable by a contract that is recognized as a valid "BlocklockContract".
+    ///      Once the decryption key is received, it triggers the internal function `_onBlocklockReceived` to handle the processing.
+    /// @param requestID The unique identifier of the blocklock request.
+    /// @param decryptionKey The decryption key that will be used to decrypt the associated ciphertext.
+    /// @notice Emits an event or performs additional logic in `_onBlocklockReceived`.
+    function receiveBlocklock(uint256 requestID, bytes calldata decryptionKey) external virtual onlyBlocklockContract {
+        _onBlocklockReceived(requestID, decryptionKey);
+    }
+
+    /// @notice Sets the Randamu subscription ID used for conitional encryption oracle services.
+    /// @dev Only callable by the contract owner.
+    /// @param subId The new subscription ID to be set.
+    function setSubId(uint256 subId) external onlyOwner {
+        subscriptionId = subId;
+        emit NewSubscriptionId(subId);
+    }
+
+    /// @notice Sets the address of the IBlocklockSender contract.
+    /// @dev Only the contract owner can call this function.
+    /// @param _blocklock The address of the deployed IBlocklockSender contract.
+    function setBlocklock(address _blocklock) external onlyOwner {
+        require(_blocklock != address(0), "Cannot set zero address as sender");
+        blocklock = IBlocklockSender(_blocklock);
+    }
+
+    /// @notice Adds a list of consumer addresses to the Randamu subscription.
+    /// @dev Requires the subscription ID to be set before calling.
+    /// @param consumers An array of addresses to be added as authorized consumers.
+    function updateSubscription(address[] calldata consumers) external onlyOwner {
```

Comment:
Is the intention that users will create a single `AbstractBlocklockReceiver` and use it to fan out requests to all their other contracts?

Reply:
If so, it probably needs a detailed example of how this would work